### ISSUE TITLE
Inject fixtures from conftest.py

### ISSUE THREAD

**Issue by calina-c** [tmarktaylor/phmdoctest#27]
Hello! I might be doing something wrong, but it seems that all the fixtures I define in conftest.py for my generated tests, including those marked as `autouse=True`, are being ignored when I run the tests. I assume this is because of the namespace manipulation or something similar. Is there any way to keep these fixtures? Is there any way to pass the fixtures I want to keep, maybe as an option to the generation command?

My use case is the following, please skip if irrelevant: there is a "basics" readme that helps users create some key entities, but the main readmes only reference it. Using the `setup` option does not work, since the code block is not in the same file. I thought about using some classic conftest fixtures but I see they are simply lost.

Right now, for example I have the following code generated by phmdoctest:
`def test_code_66(managenamespace):`

If I manually edit it to 
`def test_code_66(managenamespace, myfixture_from_conftest):`
then it works just as expected.

I do not want to setup a preliminary block or something similar, I just want access to my "regular" fixtures on top of the ones phmdoctest introduces. Can you provide some help? Are these intentional design specs or is it a bug?

Thank you in advance!

**Comment by tmarktaylor** [comment]
Hi,

It is a great idea to inject fixtures into phmdoctest generated test cases. It has the potential to allow scope="session" setup for generated tests. Currently phmdoctest setup can only be done at scope="module" for Python code/expected output fenced code blocks. (Please ignore for the moment phmdoctest for Python interactive sessions is actually at scope="session"; You aren't testing Python interactive sessions?).

I haven't tried any code yet. It is hard to imagine why it was merely "ignored" rather than throwing an exception.

`def test_code_66(managenamespace, myfixture_from_conftest):` could have the side effect of copying a reference to myfixture_from_conftest up to the module.

Try this: `python -m pytest --log-cli-level=debug --color=yes doc/test_directive3.py` and look for ` manager- adding=` to see what got copied up to the module level (listed after namespace=).

You also might try using scope="session" in conftest.py on the fixture just to see what happens.

If possible, please provide some more detail like the body of def test_code_66, the fixture source from conftest.py, and what "ignored" means in terms of python code and object values.

**Comment by calina-c** [comment]
Thank you for replying so promptly. It's ignored in the sense where the variable is undefined if I try using it, so it does throw an exception. I was using the word "ignored" as in it's not passed through to my code.

In the meantime, I explored other options for libraries and since this is a work task, I'll focus on finishing up. I'll be back with more details by the end of the week, though, since I started this discussion :)

**Comment by tmarktaylor** [comment]
I withdraw my request for more information.

I think I was able to duplicate the "ignored fixture". I don't understand autouse fixtures well enough to explain why the fixture resolves to a function. My un-informed guess is that autouse is suitable for configuring other fixtures but not for supplying values to test cases.

I made a couple of changes to the Pytest 7.1.x docs example just below "Autouse fixtures (fixtures you don’t have to request)" and it still works.
- delete parameter first_entry from fixture order.
- delete return keyword from append_first (the autouse fixture) and keep the rest of the line.
